src/main/java/stashy/parser/Parser.java

```diff
@@ -0,0 +1,317 @@
+package stashy.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+import stashy.commands.AddDeadlineCommand;
+import stashy.commands.AddEventCommand;
+import stashy.commands.AddTodoCommand;
+import stashy.commands.Command;
+import stashy.commands.DeleteCommand;
+import stashy.commands.ExitCommand;
+import stashy.commands.FindCommand;
+import stashy.commands.ListCommand;
+import stashy.commands.MarkCommand;
+import stashy.commands.UnmarkCommand;
+import stashy.data.exception.StashyException;
+import stashy.data.task.Deadline;
+import stashy.data.task.Event;
+import stashy.data.task.Task;
+import stashy.data.task.ToDo;
+
+/**
+ * Represents the parser of the user input command.
+ */
+public class Parser {
+    public static final Pattern TASK_DATA_FORMAT =
+        Pattern.compile("^\\[(?<category>[TDE])\\]"
+                + "\\[(?<isDone>[ X])\\]"
+                + " (?<description>[^\\(\\)\\n]*)\\(?"
+                + "(?<by>(by: [^\\(\\)]*)?)"
+                + "(?<at>(at: [^\\(\\)]*)?)\\)?$");
+
+    public static final String[] ACCEPTABLE_DATETIME_FORMATS = {

```

Comment:
I like how you accept multiple multiple formats for reading date time and extracted them into a constant array👍

Reply:
Thanks! I feel this is an ad-hoc solution to detect multiple date formats but it actually makes the parser more readable 😅